### PR TITLE
Rename Namespace DomainCreators to domain::creators

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Rename the namespace `DomainCreators` to `domain::creators`
- Rename `Unit.Domain.DomainCreators` to `Unit.Domain.Creators`
- Rename `DomainCreators` directories to `Creators`

This does some of #814 to separate the process and make it easier to review.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by fmahebert — DISMISSED** [review]

File: src/Domain/Creators/DomainCreator.hpp [hunk 1]
```diff
@@ -22,8 +22,9 @@ template <size_t, typename>
 class Domain;
 /// \endcond
 
+namespace domain {
```
Comment (hrueter) [inline src/Domain/Creators/DomainCreator.hpp:25]: This namespace should be documented, probably in a future PR.
A good place could be `Domain/Domain.hpp`.

**Review by hrueter — DISMISSED** [review]
group review by @nilsleiffischer and @hrueter

File: src/Domain/Creators/DomainCreator.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/Domain/Creators/DomainCreator.hpp:25]: Yep, that's the plan :)

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@wthrowe this should be an easy one to review if you have some time :)

**Comment by wthrowe** [comment]
Needs a rebase onto develop, but otherwise looks fine.

**Comment by nilsdeppe** [comment]
Restarted the GCC6 release and debug because the two-index constraint test failed. This seems to be a fairly annoying bug (#1227). All other tests passed on both.

**Review by wthrowe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Had to rebase on develop again to resolve conflicts...

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**